src/main/java/de/tum/in/www1/artemis/config/migration/setups/localvc/jenkins/MigrationEntryJenkinsToLocalVC.java

```diff
@@ -0,0 +1,142 @@
+import de.tum.in.www1.artemis.exception.JenkinsException;
+import de.tum.in.www1.artemis.repository.AuxiliaryRepositoryRepository;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseRepository;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseStudentParticipationRepository;
+import de.tum.in.www1.artemis.repository.SolutionProgrammingExerciseParticipationRepository;
+import de.tum.in.www1.artemis.repository.TemplateProgrammingExerciseParticipationRepository;
+import de.tum.in.www1.artemis.service.UriService;
+import de.tum.in.www1.artemis.service.connectors.jenkins.build_plan.JenkinsBuildPlanService;
+
+/**
+ * Migration of Jenkins Build plans when the repositories will move to LocalVC.
+ */
+@Component
+@Profile("jenkins & " + PROFILE_MIGRATE_GITLAB_JENKINS_TO_LOCALVC)
+public class MigrationEntryJenkinsToLocalVC extends LocalVCMigrationEntry {
+
+    private static final Logger log = LoggerFactory.getLogger(MigrationEntryJenkinsToLocalVC.class);
+
+    private final JenkinsBuildPlanService jenkinsBuildPlanService;
+
+    /**
+     * Base url to the local VC repository is the server urls plus /git/.
+     */
+    private String localVCRepositoriesBaseUrl;
+
+    @Value("${artemis.version-control.url}")
+    private String sourceVCSBaseUrl;
+
+    /**
+     * URL to the source VCS with a trailing slash.
+     * This avoids that a base URL like "http://example.com" also finds "http://example.com:8080".
+     */
+    private String sourceVCSRepositoriesBaseUrl;
+
+    public MigrationEntryJenkinsToLocalVC(ProgrammingExerciseRepository programmingExerciseRepository,
+            SolutionProgrammingExerciseParticipationRepository solutionProgrammingExerciseParticipationRepository,
+            TemplateProgrammingExerciseParticipationRepository templateProgrammingExerciseParticipationRepository,
+            ProgrammingExerciseStudentParticipationRepository programmingExerciseStudentParticipationRepository, UriService uriService,
+            JenkinsBuildPlanService jenkinsBuildPlanService, AuxiliaryRepositoryRepository auxiliaryRepositoryRepository) {
+        super(programmingExerciseRepository, solutionProgrammingExerciseParticipationRepository, templateProgrammingExerciseParticipationRepository,
+                programmingExerciseStudentParticipationRepository, auxiliaryRepositoryRepository);
+        this.jenkinsBuildPlanService = jenkinsBuildPlanService;
+    }
+
+    @PostConstruct
+    public void initialize() {
+        localVCRepositoriesBaseUrl = localVCBaseUrl + "/git/";
+        sourceVCSRepositoriesBaseUrl = sourceVCSBaseUrl.endsWith("/") ? sourceVCSBaseUrl : sourceVCSBaseUrl + "/";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected Logger getLogger() {
+        return log;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void migrateSolutions(List<SolutionProgrammingExerciseParticipation> solutionParticipations) {
```

Comment:
Not sure, but do we also need to change the URIs of auxiliary repositories?

Reply:
You are right, here is no explicit handling for the auxilary repositories necessary, because there are no build plans for them. The JavaDoc from the super class was wrong there. :heavy_check_mark: I fixed that with b53d29c3766907c7f14973862d8deb7bb5f5f7fc